int/sni/sni.go

```diff
@@ -0,0 +1,171 @@
+	}
+
+	template := &x509.Certificate{
+		SerialNumber: new(big.Int).SetBytes(uniqueSiteNameID),
+		Subject: pkix.Name{
+			CommonName:   serverName,
+			Organization: []string{"station dynamically generated"},
+		},
+		NotBefore: time.Now(),
+		NotAfter:  time.Now().AddDate(0, 0, 1),
+		DNSNames:  []string{serverName},
+	}
+
+	return template, nil
+}
+
+// generateSignedCertificate creates a certificate and then signs it using the provided Certificate Authority (CA).
+// It uses root certificate and private key from the default location.
+// It uses hashServerName and createCertificateTemplate to ensure uniqueness and proper formatting of the certificate.
+func generateSignedCertificate(serverName string) ([]byte, *rsa.PrivateKey, error) {
+	privateKey, err := rsa.GenerateKey(rand.Reader, privateKeySizeInBits)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to generate private key: %w", err)
+	}
+
+	caPath, err := mkcertCARootPath()
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to get mkcert CA root path: %w", err)
+	}
+
+	caCertificate, err := certificate.LoadCertificate(filepath.Join(caPath, caCertificateFileName))
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to load CA certificate: %w", err)
+	}
+
+	caPrivateKey, err := certificate.LoadPrivateKey(filepath.Join(caPath, caPrivateKeyFileName))
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to load CA private key: %w", err)
+	}
+
+	uniqueSiteNameID, err := hashServerName(serverName)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	template, err := createCertificateTemplate(serverName, uniqueSiteNameID)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	cert, err := x509.CreateCertificate(rand.Reader, template, caCertificate, privateKey.Public(), caPrivateKey)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to create certificate: %w", err)
+	}
+
+	return cert, privateKey, nil
+}
+
+// GenerateTLS creates a TLS certificate using the server name of the client hello info request.
+// Implementation design: it logs information because log configuration at go-swagger level is not available.
+func GenerateTLS(hello *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	logger.Debugf("%s generating TLS certificate", loggerPrefix)
+
+	if hello == nil {
+		logger.Errorf("%s client hello info is nil", loggerPrefix)
+
+		return nil, fmt.Errorf("%w: client hello info is nil", ErrInvalidArgument)
+	}
+
+	certBytes, privateKey, err := generateSignedCertificate(hello.ServerName)
+	if err != nil {
+		logger.Errorf("%s generate signed certificate for %s failed: %w", loggerPrefix, hello.ServerName, err)
+
+		return nil, fmt.Errorf("%s generate signed certificate for %s failed: %w", loggerPrefix, hello.ServerName, err)
+	}
+
+	return &tls.Certificate{
+		Certificate: [][]byte{certBytes},
+		PrivateKey:  privateKey,
+	}, nil
+}
+
+// Get the mkcert CA root path depending on the OS.
+func mkcertCARootPath() (string, error) {
```

Comment:
I don't think it's a good idea because check config shouldn't be a package. In my initial PR everything was done. But it was to big. I propose to leave it like that and then to clean the config/pkg.